gpflow/base.py

```diff
@@ -73,9 +73,9 @@ def log_prior(self):
         bijector = self.transform
         if self.prior is not None:
             out += tf.reduce_sum(self.prior.log_prob(x))
-        if self.transform is not None:
-            log_det_jacobian = bijector.forward_log_det_jacobian(y, y.shape.ndims)
-            out += tf.reduce_sum(log_det_jacobian)
+            if self.transform is not None:
+                log_det_jacobian = bijector.forward_log_det_jacobian(y, y.shape.ndims)
+                out += tf.reduce_sum(log_det_jacobian)
```

Comment:
Only add log_det_jacobian if there IS a prior

Reply:
What would be a good way of testing that the log-det-Jacobian of the transform is correctly added to the objective when doing MAP? :thinking:

Reply:
Good catch! Yes, we should add the log det jacobian only if prior exists. I would even suggest that None must be returned when the prior is None.

Reply:
If we return `None` then the return type will be unstable, and all places that use this method would have to explicitly check that it didn't return None before adding it to the objective ... is that actually what we want?

Reply:
I would say that's logical. No prior - you can't return a value. `None` is a good indicator that there is no valid value for the prior of the parameter.